jackrabbit-api/src/main/java/org/apache/jackrabbit/api/JackrabbitValueFactory.java

```diff
@@ -0,0 +1,188 @@
+ *     </li>
+ * </ol>
+ * <p>
+ * <h3>Direct Binary Download</h3>
+ * <p>
+ * The direct binary download process is described in detail in {@link
+ * BinaryDownload}.
+ */
+@ProviderType
+public interface JackrabbitValueFactory extends ValueFactory {
+    /**
+     * Initiate a transaction to upload binary data directly to a storage
+     * location.  {@link IllegalArgumentException} will be thrown if an upload
+     * cannot be supported for the required parameters, or if the parameters are
+     * otherwise invalid.  For example, if the value of {@code maxSize} exceeds
+     * the size limits for a single binary upload for the implementation or the
+     * service provider, or if the value of {@code maxSize} divided by {@code
+     * maxParts} exceeds the size limit for an upload or upload part of the
+     * implementation or the service provider, {@link IllegalArgumentException}
+     * may be thrown.
+     * <p>
+     * Each service provider has specific limitations on upload sizes,
+     * multi-part upload support, part sizes, etc. which can result in {@link
+     * IllegalArgumentException} being thrown.  You should consult the
+     * documentation for your underlying implementation and your service
+     * provider for details.
+     * <p>
+     * If this call is successful, a {@link BinaryUpload} is returned
+     * which contains the information a client needs to successfully complete
+     * a direct upload.
+     *
+     * @param maxSize The expected maximum size of the binary to be uploaded by
+     *         the client.  If the actual size of the binary is known, this
+     *         size should be used; otherwise, the client should make a best
+     *         guess.  If a client calls this method with one size and then
+     *         later determines that the guess was too small, the transaction
+     *         should be restarted by calling this method again with the correct
+     *         size.
+     * @param maxURIs The maximum number of upload URIs that the client can
+     *         accept.  The implementation will ensure that an upload of size
+     *         {@code maxSize} can be completed by splitting the value of {@code
+     *         maxSize} into parts, such that the size of the largest part does
+     *         not exceed any known implementation or service provider
+     *         limitations on upload part size and such that the number of parts
+     *         does not exceed the value of {@code maxURIs}.  If this is not
+     *         possible, {@link IllegalArgumentException} will be thrown.  A
+     *         client may specify -1 for this value, indicating that any number
+     *         of URIs may be returned.
+     * @return A {@link BinaryUpload} that can be used by the client to complete
+     *         the upload via a call to {@link #completeBinaryUpload(String)},
+     *         or {@code null} if the implementation does not support the direct
+     *         upload feature.
```

Comment:
This should throw an exception instead of returning null.

(should we also define a repository descriptor maybe?)